### ISSUE TITLE
Test failures are swallowed by MockitoFixture

### ISSUE THREAD

**Issue by Krever** [mockito/mockito-scala#33]
Current implementation:
```
  override def withFixture(test: NoArgTest): Outcome = {
    val session = MockitoScalaSession(name = s"MockitoFixtureSession[${test.name}]")
    val outcome = super.withFixture(test)
    session.finishMocking()
    outcome
  }
```
If `outcome` is a failed test and we have unused stubbings because of it then `session.finishMocking()` will throw an exception and the original test failure wont be presented to the user.

**Comment by bbonanno** [comment]
PR #30 addresses this (I hope, maybe I missed it altogether, but I've done a lot of changes there) and some other issues/improvements, please take a look at it if you can, is not ready yet but I raised it to collect people's opinions

**Comment by Krever** [comment]
Sure, thanks!

**Comment by bbonanno** [comment]
Fixed on #30
